contracts/volt/VoltV2.sol

```diff
@@ -0,0 +1,435 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {VoltCoreRef} from "../refs/VoltCoreRef.sol";
+
+// Forked from Uniswap's UNI
+// Reference: https://etherscan.io/address/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984#code
+
+contract VoltV2 is VoltCoreRef {
+    using SafeCast for *;
+
+    /// @notice EIP-20 token name for this token
+    // solhint-disable-next-line const-name-snakecase
+    string public constant name = "Volt";
+
+    /// @notice EIP-20 token symbol for this token
+    // solhint-disable-next-line const-name-snakecase
+    string public constant symbol = "VOLT";
+
+    /// @notice EIP-20 token decimals for this token
+    // solhint-disable-next-line const-name-snakecase
+    uint8 public constant decimals = 18;
+
+    /// @notice Total number of tokens in circulation
+    // solhint-disable-next-line const-name-snakecase
+    uint256 public totalSupply;
+
+    /// @notice Allowance amounts on behalf of others
+    mapping(address => mapping(address => uint256)) internal allowances;
+
+    /// @notice Official record of token balances for each account
+    mapping(address => uint256) internal balances;
+
+    /// @notice A record of each accounts delegate
+    mapping(address => address) public delegates;
+
+    /// @notice A checkpoint for marking number of votes from a given block
+    struct Checkpoint {
+        uint32 fromBlock;
+        uint224 votes;
```

Comment:
include explanation as to why this 224 for votes and 256 for balances is acceptable.

Reply:
can be copy pasted from oz's explanation on their erc20 votes

Reply:
with attribution